Homework5/ISBNbook/Models/Catalogue.cs

```diff
@@ -0,0 +1,69 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
+
+namespace ISBNbook.Models
+{
+    public class Catalogue
+    {
+        private Dictionary<string, Book> BookCollection = new Dictionary<string, Book>();
+
+
+        public void AddBook(Book book, string isbn)
+        {
+            if (IsFormatCorrect(isbn))
+            {
+                BookCollection.Add(ToDefaultFormat(isbn), book);
```

Comment:
The code will throw exception when the book with the same isbn is added, consider to update the existing book intstead